rotkehlchen/tests/api/test_balances.py

```diff
@@ -911,3 +912,35 @@ def test_balances_behaviour_with_manual_current_prices(rotkehlchen_api_server, e
         rdn_result = result['assets']['eip155:1/erc20:0x255Aa6DF07540Cb5d3d297f0D0D4D84cb52bc8e6']
         assert rdn_result['amount'] == '5'
         assert rdn_result['usd_value'] == '150.0'
+
+
+@pytest.mark.parametrize('number_of_eth_accounts', [1])
+def test_blockchain_balances_refresh(
+        rotkehlchen_api_server: 'APIServer',
+        ethereum_accounts,
+):
+    """Checks that blockchain balances are refreshed properly when the endpoint is called"""
+    rotki = rotkehlchen_api_server.rest_api.rotkehlchen
+
+    account_balance = {ethereum_accounts[0]: BalanceSheet(
+        assets=defaultdict(Balance, {
+            A_USDC: Balance(FVal(1), FVal(24)),
+            A_DAI: Balance(FVal(2), FVal(42)),
+            A_USDT: Balance(FVal(3), FVal(54)),
+        }),
+    )}
+    account_balance_patch = patch.object(rotki.chains_aggregator.balances, 'eth', account_balance)
+    mock_balances = {ethereum_accounts[0]: {A_USDC: FVal(23), A_DAI: FVal(3)}}
+    mock_prices = {A_USDC: Price(FVal(10)), A_DAI: Price(FVal(11))}
+    query_tokens_patch = patch('rotkehlchen.chain.evm.tokens.EvmTokens.query_tokens_for_addresses', side_effect=lambda addresses: (mock_balances, mock_prices))  # noqa: E501
+    with account_balance_patch, query_tokens_patch:
+        response = requests.get(api_url_for(
+            rotkehlchen_api_server,
+            'blockchainbalancesresource',
+        ), json={'async_query': False, 'blockchain': 'ETH'})
+        result = assert_proper_response_with_result(response)
+        assert result['per_account']['eth'][ethereum_accounts[0]]['assets'] == {
+            'ETH': {'amount': '0', 'usd_value': '0.0'},
+            A_USDC.resolve_to_evm_token().identifier: {'amount': '23', 'usd_value': '230'},
+            A_DAI.resolve_to_evm_token().identifier: {'amount': '3', 'usd_value': '33'},
```

Comment:
you don't need to resolve to get the identifier. The identifier is in teh `Asset` class, is the only thing that they have

```suggestion
            A_USDC.identifier: {'amount': '23', 'usd_value': '230'},
            A_DAI.identifier: {'amount': '3', 'usd_value': '33'},
```